service/framework.go

```diff
@@ -337,8 +338,18 @@ func newCRDFramework(config Config) (*framework.Framework, error) {
 	}
 
 	initCtxFunc := func(ctx context.Context, obj interface{}) (context.Context, error) {
-		if config.Viper.GetBool(config.Flag.Service.Guest.Update.Enabled) {
-			updateallowedcontext.SetUpdateAllowed(ctx)
+		{
+			customObject, err := keyv3.ToCustomObject(obj)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			updateEnabled := config.Viper.GetBool(config.Flag.Service.Guest.Update.Enabled)
+			versionBundleVersion := keyv3.VersionBundleVersion(customObject)
+
+			if updateEnabled && versionBundleVersion >= "1.1.0" {
+				updateallowedcontext.SetUpdateAllowed(ctx)
+			}
```

Comment:
When updates are enabled and the version bundle version is at least the current latest one we modify the context for further use down the road.